warehouse/query-core/src/main/java/datawave/query/util/FuzzyAttributeComparator.java

```diff
@@ -0,0 +1,91 @@
+package datawave.query.util;
+
+import com.google.common.collect.Sets;
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.Attributes;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public final class FuzzyAttributeComparator {
+    
+    // this is a utility and should never be instantiated
+    private FuzzyAttributeComparator() {
+        throw new UnsupportedOperationException();
+    }
+    
+    public static boolean singleToSingle(final Attribute existingAttribute, final Attribute newAttribute) {
+        return (existingAttribute.getData().equals(newAttribute.getData())
+                        && existingAttribute.getColumnVisibility().equals(newAttribute.getColumnVisibility()) && existingAttribute.getTimestamp() == newAttribute
+                        .getTimestamp());
+    }
+    
+    public static boolean singleToMultiple(final Attributes multipleAttributes, final Attribute singleAttribute) {
+        return multipleAttributes.getAttributes().stream().anyMatch(existingAttribute -> {
+            return singleToSingle(existingAttribute, singleAttribute);
+        });
+    }
+    
+    public static boolean multipleToMultiple(final Attributes existingAttributes, final Attributes newAttributes) {
+        boolean containsMatch = false;
+        for (Attribute<? extends Comparable<?>> newAttr : newAttributes.getAttributes()) {
+            if (singleToMultiple(existingAttributes, newAttr)) {
+                containsMatch = true;
+            }
+        }
+        return containsMatch;
+    }
+    
+    /**
+     * Performs a basic fuzzy match on the content/data of two given Attributes. If such a match is found, combine the metadata from the two Attributes and
+     * create a new set of deduplicated Attributes to return to the Document.
+     *
+     * @param existingAttributes
+     *            The set of Attributes against which we want to compare
```

Comment:
Strictly speaking, we do not have a set of attributes (`Set<Attributes>`). We have a single instance of `Attributes`, which is itself a set of type attribute (`Set<Attribute>`)